src/api/services/ProfileService.ts

```diff
@@ -0,0 +1,119 @@
+import * as Bookshelf from 'bookshelf';
+import { inject, named } from 'inversify';
+import { Logger as LoggerType } from '../../core/Logger';
+import { Types, Core, Targets } from '../../constants';
+import { validate, request } from '../../core/api/Validate';
+import { NotFoundException } from '../exceptions/NotFoundException';
+import { ProfileRepository } from '../repositories/ProfileRepository';
+import { Profile } from '../models/Profile';
+import { ProfileCreateRequest } from '../requests/ProfileCreateRequest';
+import { ProfileUpdateRequest } from '../requests/ProfileUpdateRequest';
+import { RpcRequest } from '../requests/RpcRequest';
+import { AddressService } from './AddressService';
+
+
+export class ProfileService {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Service) @named(Targets.Service.AddressService) public addressService: AddressService,
+        @inject(Types.Repository) @named(Targets.Repository.ProfileRepository) public profileRepo: ProfileRepository,
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+    ) {
+        this.log = new Logger(__filename);
+    }
+
+
+    @validate()
+    public async rpcFindAll( @request(RpcRequest) data: any): Promise<Bookshelf.Collection<Profile>> {
+        return this.findAll();
+    }
+
+    public async findAll(withRelated: boolean = true): Promise<Bookshelf.Collection<Profile>> {
+        return this.profileRepo.findAll(withRelated);
+    }
+
+    @validate()
+    public async rpcFindOne( @request(RpcRequest) data: any): Promise<Profile> {
+        return this.findOne(data.params[0]);
+    }
+
+    public async findOne(id: number, withRelated: boolean = true): Promise<Profile> {
+        const profile = await this.profileRepo.findOne(id, withRelated);
+        if (profile === null) {
+            this.log.warn(`Profile with the id=${id} was not found!`);
+            throw new NotFoundException(id);
+        }
+        return profile;
+    }
+
+    @validate()
```

Comment:
params MUST be an Array, containing the values in the Server expected order.
so, it would be quite hard to implement rpcSaveProfile that could save unknown amount of Addresses.
which means we just need to create a profile to the user and then use rpcSaveAddress to save addresses to the profile.

we also need rpcGetProfile to fetch all the Profile related data.

most likely we are always saving information to only one profile, but still we could add a name field to it for future. make it unique and pass it as the only parameter to rpcGetProfile and rpcSaveProfile.